playwright-test/harness.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    test as base,
+    chromium,
+    type BrowserContext,
+    type Worker,
+    type Route,
+    type Page,
+    type Request,
+} from "@playwright/test";
+import path from "path";
+import fs from "fs/promises";
+
+function getManifestVersion() {
+    return process.env.npm_lifecycle_event === "playwright-mv3" ? 3 : 2;
+}
+
+async function routeLocalResources(route: Route) {
+    const url = new URL(route.request().url());
+    const localPath = path.join(__dirname, "data", url.pathname);
+    try {
+        const body = await fs.readFile(localPath);
+        console.log("request served from disk", route.request().url());
+        route.fulfill({ status: 200, body });
+    } catch (e) {
+        console.log("request served from network", route.request().url());
+        route.continue();
+    }
+}
+
+export const mockAtb = {
+    majorVersion: 364,
+    minorVersion: 2,
+    version: 'v364-2'
+};
+
+// based off example at https://playwright.dev/docs/chrome-extensions#testing
+export const test = base.extend<{
+    manifestVersion: 2 | 3;
+    context: BrowserContext;
+    backgroundPage: Page | Worker;
+    routeExtensionRequests: (url: string | RegExp, handler: (route: Route, request: Request) => any) => Promise<void>;
+    backgroundNetworkContext: Page | BrowserContext;
+}>({
+    manifestVersion: getManifestVersion(),
+    context: async ({ manifestVersion }, use) => {
+        const extensionPath =
+            manifestVersion === 3 ? "build/chrome-mv3/dev" : "build/chrome/dev";
+        const pathToExtension = path.join(__dirname, "..", extensionPath);
+        const context = await chromium.launchPersistentContext("", {
+            headless: false,
+            args: [
+                `--disable-extensions-except=${pathToExtension}`,
+                `--load-extension=${pathToExtension}`,
+            ],
+        });
+        // intercept extension install page and use HAR
+        context.on("page", (page) => {
+            console.log("page", page.url());
+            if (page.url().includes("duckduckgo.com/extension-success")) {
+                page.routeFromHAR(path.join(__dirname, "data", "postinstall.har"), {
+                    notFound: "abort",
+                });
+            }
+        });
+        //
+        await use(context);
+        await context.close();
+    },
+    backgroundPage: async ({ context, manifestVersion }, use) => {
```

Comment:
We generally use the following structure:

```suggestion
    async backgroundPage ({ context, manifestVersion }, use) {
```

Reply:
Sorry, I dunno why I decided to give other feedback (I hadn't noticed it was a PoC 👍🏻 )

Reply:
While it's a Nit, I do agree. Please could you use the method shorthand for these functions @sammacbeth?